test/extensions/filters/http/ext_proc/state_test.cc

```diff
@@ -11,75 +11,165 @@ namespace {
 
 TEST(StateTest, EmptyQueue) {
   ChunkQueue queue;
+  Buffer::OwnedImpl received_data;
+  Buffer::OwnedImpl out_data;
   EXPECT_TRUE(queue.empty());
   EXPECT_EQ(0, queue.bytesEnqueued());
-  EXPECT_FALSE(queue.pop(false));
+  EXPECT_FALSE(queue.pop(false, received_data, out_data));
 }
 
 TEST(StateTest, BasicQueue) {
   ChunkQueue queue;
+  Buffer::OwnedImpl received_data;
+  Buffer::OwnedImpl out_data;
   Buffer::OwnedImpl data1("Hello");
-  queue.push(data1, false, false);
+
+  EXPECT_EQ(received_data.length(), 0);
+  queue.push(data1, false, false, received_data);
+  EXPECT_EQ(received_data.toString(), "Hello");
   EXPECT_FALSE(queue.empty());
   EXPECT_EQ(5, queue.bytesEnqueued());
-  auto popped = queue.pop(false);
-  EXPECT_EQ((*popped)->data.toString(), "Hello");
+  auto popped = queue.pop(false, received_data, out_data);
+  EXPECT_EQ(out_data.toString(), "Hello");
+  EXPECT_FALSE((*popped)->end_stream);
+  EXPECT_FALSE((*popped)->delivered);
+  EXPECT_EQ((*popped)->buffer_length, 5);
   EXPECT_TRUE(queue.empty());
   EXPECT_EQ(0, queue.bytesEnqueued());
+  EXPECT_EQ(received_data.length(), 0);
+}
+
+TEST(StateTest, EmptyString) {
```

Comment:
Can this be given a better name? It's really hard to tell what this is trying to test with the given name